gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/UserQuotaManager.java

```diff
@@ -41,39 +42,50 @@
 @Slf4j
 public class UserQuotaManager {
   public static final String PER_USER_QUOTA = DagManager.DAG_MANAGER_PREFIX + "perUserQuota";
+  public static final String PER_FLOWGROUP_QUOTA = DagManager.DAG_MANAGER_PREFIX + "perFlowGroupQuota";
   public static final String USER_JOB_QUOTA_KEY = DagManager.DAG_MANAGER_PREFIX + "defaultJobQuota";
   public static final String QUOTA_SEPERATOR = ":";
   public static final Integer DEFAULT_USER_JOB_QUOTA = Integer.MAX_VALUE;
   private final Map<String, Integer> proxyUserToJobCount = new ConcurrentHashMap<>();
+  private final Map<String, Integer> flowGroupToJobCount = new ConcurrentHashMap<>();
   private final Map<String, Integer> requesterToJobCount = new ConcurrentHashMap<>();
   private final Map<String, Integer> perUserQuota;
+  private final Map<String, Integer> perFlowGroupQuota;
   Map<String, Boolean> runningDagIds = new ConcurrentHashMap<>();
   private final int defaultQuota;
 
   UserQuotaManager(Config config) {
     this.defaultQuota = ConfigUtils.getInt(config, USER_JOB_QUOTA_KEY, DEFAULT_USER_JOB_QUOTA);
-    ImmutableMap.Builder<String, Integer> mapBuilder = ImmutableMap.builder();
-
+    ImmutableMap.Builder<String, Integer> userMapBuilder = ImmutableMap.builder();
+    ImmutableMap.Builder<String, Integer> flowGroupMapBuilder = ImmutableMap.builder();
+    // Quotas will take form of user:flowGroup:<Quota>
```

Comment:
Am I interpreting this correctly? Each user has a quota (let's call it X) that can be applied across flowGroups. X can be distributed among flows for different groups or all go to one flow group's flows? At a higher level each flowGroup also has a max quota that is enforced, which is why the user quota check occurs prior to the group level check in the code below. It would be useful to explain this in a comment above the checkQuota function or to explain this mapping better here.

Reply:
Oh let me update this, this was an initial design but I changed the design but forgot to change the comment.

Quotas should not correlate users and flowgroups together, since a single flowgroup can have multiple user submissions and a single user can also submit to multiple quotas. We should configure these individually.